React/filter box refactor

clairesunstudio opened the pull request:
- (React) DP-1234: Refactor FilterBox to allow any form components 
Note: This will need to be followed up in search and other apps consuming FilterBox / SearchBanner

This is a step towards making searchBanner with postInputFilter responsive on mobile

clairesunstudio committed:
wip

clairesunstudio committed:
wip

clairesunstudio committed:
clean up all conditional rendering in story

clairesunstudio committed:
clean up filterbox index

clairesunstudio committed:
clean up searchbanner postinputfilter data

clairesunstudio committed:
clean up searchbanner

clairesunstudio committed:
wip

clairesunstudio committed:
clean up

clairesunstudio committed:
space out filters fields

clairesunstudio committed:
add changelog

clairesunstudio committed:
update screenshots

clairesunstudio committed:
update changelog

pmallett-mc commented on react/src/components/organisms/FilterBox/index.js:
```diff
@@ -25,27 +23,11 @@ const FilterBox = (props) => {
-                <div className="ma__filter-box__organizations">
-                  <InputTextTypeAhead {...organization} />
+              { fields.map((field) => (
+                <div className={field.class}>
```
these will need a `key` attribute now

pmallett-mc reviewed (changes requested):
This looks pretty good, but the FilterBox is failing to render in Storybook when I test this in conjunction w/ the latest develop:
![filterbox-errors](https://user-images.githubusercontent.com/38226003/46176276-1b97f480-c26c-11e8-8535-98ecfceaa3bb.png)

Also, though SearchBanner functions, there are a lot of console warnings, mostly "Warning: Encountered two children with the same key"

I also think this should be tested out with Search & we should have a PR to go in with it, after all, if nothing can use this, there's no reason to rush into it.

clairesunstudio committed:
add unique keys to filter fields

clairesunstudio committed:
fix filterbox story

clairesunstudio commented:
@petermallett good catch! updated and fixed filterbox. This is part of the search banner work I'm doing for mobile responsiveness for EEC. I separated this out so that it's easier to review. I created a pr in search just so we don't forget, but it won't work until https://github.com/massgov/massgov-search/pull/326 and https://github.com/massgov/massgov-search/pull/327 get in so that we can bring in the new mayflower. We should spend some time next week to get those resolved

clairesunstudio commented:
But for now, @petermallett can you review this again so that i can continue working on search banner styles for EEC.  created a test branch here -->https://github.com/massgov/eec-childcare-provider-search/pull/125 thanks!

clairesunstudio committed:
export forms styles to fix input styles

smurrayatwork committed:
Adds validation for filters, puts back in filter loop.

smurrayatwork committed:
Adds comments to utility function usage.

clairesunstudio committed:
lint fix

clairesunstudio commented on react/src/components/utilities/componentPropTypeCheck.js:
```diff
@@ -6,9 +9,73 @@ const componentPropTypeCheck = (props, propName, componentName, componentString)
+      if (!child.component) {
+        return false;
+      }
+      if (typeof child.component.type === 'string') {
```
do you mean `child.component.type.name` here?

clairesunstudio commented on react/src/components/utilities/componentPropTypeCheck.js:
```diff
@@ -6,9 +9,73 @@ const componentPropTypeCheck = (props, propName, componentName, componentString)
+      if (typeof child.component.type === 'string') {
+        return componentTypes.indexOf(child.component.type) > -1;
+      }
+      return child.component.type.name &&
```
this seems repetitive to the one above

clairesunstudio commented on react/src/components/utilities/componentPropTypeCheck.js:
```diff
@@ -6,9 +9,73 @@ const componentPropTypeCheck = (props, propName, componentName, componentString)
+    });
+  };
+  if (!component || (component && !isValid(component))) {
+    return new Error(`Invalid prop ${propName} supplied to ${componentName}. Got: ${component || 'undefined'}. Validation failed.`);
```
the error message doesn't return the component name correctly

clairesunstudio commented:
error message example:
![screen shot 2018-09-28 at 11 19 55 am](https://user-images.githubusercontent.com/5789411/46219218-be0bb280-c314-11e8-8dc0-0d8ffed7a619.png)

clairesunstudio committed:
add comments

clairesunstudio committed:
remove unused lines in typecheck
